pegen/testutil.py

```diff
@@ -76,6 +77,9 @@ def generate_parser_c_extension(
     Returns a module object with a parse_string() method.
     TODO: express that using a Protocol.
     """
+    # Check that the working directory is empty: reusing non-empty temporary
+    # directories when generating extensions can lead to segmentation faults.
+    assert not os.listdir(path)
```

Comment:
This does seem like an awfully crude check. Maybe checking that a specific file isn't present will work? (But then, why not delete it if it exists rather than insist it is deleted?)

Reply:
> Maybe checking that a specific file isn't present will work? (But then, why not delete it if it exists rather than insist it is deleted?)

I need to understand better then what file exactly is causing the corrupted extension. Once I understand that we can delete that specific set of files. I was only checking that the temporary directory is empty to make sure that we don't add more tests that potentially segfault in the meantime.